src/Connection.php

```diff
@@ -154,77 +149,72 @@ public static function connect($dsn, $user = null, $password = null, $args = [])
-            case 'pgsql':
-                $c = new Postgresql\Connection(array_merge([
-                    'connection' => static::getPDO($dsn),
-                    'driverType' => $dsn['driverType'],
-                ], $args));
-
-                break;
-            case 'dumper': // deprecated - will be removed in dec-2020
-            case 'stopwatch':
-                $c = new Debug\Stopwatch\Connection(array_merge([
-                    'connection' => static::connect($dsn['rest'], $dsn['user'], $dsn['pass']),
-                ], $args));
-
-                break;
-            case 'counter': // deprecated - will be removed in dec-2020
-            case 'profile':
-                $c = new Debug\Profiler\Connection(array_merge([
-                    'connection' => static::connect($dsn['rest'], $dsn['user'], $dsn['pass']),
-                ], $args));
-
-                break;
-            // let PDO handle the rest
-            default:
-                $c = new static(array_merge([
-                    'connection' => static::connect(static::getPDO($dsn)),
-                ], $args));
+        /**
+         * @var Connection $connectionClass
+         */
+        $connectionClass = self::resolveConnection($dsn['driverType']);
+
+        return new $connectionClass(array_merge([
+            'connection' => $connectionClass::establishConnection($dsn),
+        ], $args));
+    }
+
+    /**
+     * Adds connection class to the registry for resolving in Connection::resolve method.
+     *
+     * Can be used as:
+     *
+     * Connection::register('mysql', MySQL\Connection::class), or
+     * MySQL\Connection::register()
+     *
+     * CustomDriver\Connection must be descendant of Connection class.
+     *
+     * @param string $connectionType
+     * @param string $connectionClass
+     */
+    public static function registerConnection($connectionType = null, $connectionClass = null)
+    {
+        if (!$connectionClass && is_a($connectionType, self::class, true)) {
+            $connectionClass = $connectionType;
+            $connectionType = null;
+        }
+
+        $connectionClass = $connectionClass ?? static::class;
+
+        $connectionType = $connectionType ?? $connectionClass::defaultDriverType();
+
+        if (is_array($driverTypes = $connectionType)) {
+            foreach ($driverTypes as $connectionType => $connectionClass) {
+                if (is_numeric($connectionType)) {
+                    $connectionType = $connectionClass::defaultDriverType();
+                }
+
+                static::registerConnection($connectionType, $connectionClass);
+            }
         }
 
-        return $c;
+        self::$connectionRegistry[$connectionType] = $connectionClass;
+    }
+
+    /**
+     * Resolves the connection class to use based on driver type.
+     *
+     * @param string $connectionType
+     *
+     * @return string
+     */
+    public static function resolveConnection($connectionType)
```

Comment:
can we refactor whole registry from `Connection` to `ConnectionClass`?

Reply:
We can `registry` designates what it is for. Can be also `connectionClassRegistry` if more clear

Reply:
yes, much better

Reply:
you renamed only the properly, not related methods (`registerConnection`, `resolveConnection`)